clippy_dev/src/update_lints.rs

```diff
@@ -894,7 +894,7 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_parse_contents() {
```

Comment:
Since this is a `restriction` lint for the time being, and those are not applied automatically to Clippy sources, I would prefer that you keep this PR limited to the lint itself, and not modify other files at the same time.

Once this is done, can you please squash the commit to get only one with the new lint?

Reply:
Good point --  reverted those changes (and squashed everything into a single commit).